src/console_cmds.cpp

```diff
@@ -1872,6 +1873,51 @@ DEF_CONSOLE_CMD(ConNewGRFReload)
 	return true;
 }
 
+DEF_CONSOLE_CMD(ConNewGRFProfile)
+{
+	if (argc == 0) {
+		IConsoleHelp("Collect a callback profiling session from a NewGRF for a number of in-game days.");
+		IConsoleHelp("Usage: newgrf_profile [<grf-num> <num-days>]");
+		IConsoleHelp("Run without parameters to get a list of active NewGRFs and their grf-num");
+		return true;
```

Comment:
If debugging multiple grfs would be a thing, one could imagine these commands:
* Show status: newgrf_profile
* Add GRF to profiler: newgrf_profile enable <grf-num>
* Remove GRF from profiler: newgrf_profile disable <grf-num>
* Run profiler for N days: newgrf_profile start <days>
* Run profiler indefinitely: newgrf_profile start
* Stop profiler: newgrf_profile stop